server/index.js

```diff
@@ -69,6 +69,118 @@ app.get('/api/products/:productId', (req, res, next) => {
     });
 });
 
+app.get('/api/cart', (req, res, next) => {
+  const cartId = req.session.cartId;
+  if (!cartId) {
+    return res.status(200).json([]);
+  }
+  const sql = `
+    select "c"."cartItemId",
+           "c"."price",
+           "p"."productId",
+           "p"."image",
+           "p"."name",
+           "p"."shortDescription"
+      FROM "cartItems" AS "c"
+      JOIN "products" AS "p" USING ("productId")
+     WHERE "c"."cartId" = $1
+  `;
+  const params = [cartId];
+  db.query(sql, params)
+    .then(result => {
+      const cart = result.rows[0];
+      if (!cart) {
+        next(new ClientError(`Cannot find cart with "cartId" ${cartId}`, 404));
+      } else {
+        res.status(200).json(cart);
+      }
+    })
+    .catch(err => {
+      next(err);
+    });
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = req.body.productId;
+  if (!parseInt(productId, 10)) {
+    return next(new ClientError('"productId" must be a positive integer', 400));
+  }
+
+  const sql = `
+    SELECT "price"
+      FROM "products"
+     WHERE "productId" = $1
+  `;
+  const params = [productId];
+  db.query(sql, params)
+    .then(result => {
+      const product = result.rows[0];
+      if (!product) {
+        return next(new ClientError(`Cannot find product with "productId ${productId}`, 404));
+      }

```

Comment:
You'll need to `throw` instead of `return next()` to short-circuit to your `.catch()`. Otherwise, the next `.then()` will still run.